bftengine/src/bftengine/DbCheckpointManager.cpp

```diff
@@ -236,8 +223,12 @@ void DbCheckpointManager::removeOldestDbCheckpoint() {
     dbCheckptMetadata_.dbCheckPoints_.erase(it);
     LOG_INFO(getLogger(), "removed db checkpoint, id: " << it->second.checkPointId_);
     updateDbCheckpointMetadata();
+    removeDbCheckpointFuture(it->second.checkPointId_);
   }
 }
+void DbCheckpointManager::removeDbCheckpointFuture(uint64_t id) {
+  if (auto it = dbCreateCheckPtFuture_.find(id); it != dbCreateCheckPtFuture_.end()) dbCreateCheckPtFuture_.erase(it);
```

Comment:
We could use `dbCreateCheckPtFuture_.erase(id)` by using the `std::map::erase(Key)` overload.